### DISCUSSION TITLE
Change Login Banner

### DISCUSSION THREAD

**Discussion by ghost** [microsoft/WSL#6499] [General]
Hello. I have an Kali WSL on Windows and everytime I launch it, It displays:

┏━(Message from Kali developers)
┃
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
┃
┗━(Run “touch ~/.hushlogin” to hide this message)

But I want to change it (Not remove it). I searched everywhere but I found nothing. Please help

**Comment by Biswa96** [comment]
Try to edit `/etc/motd` or `/etc/update-motd.d/*` files.

**Comment by hashara256** [comment]
In case you did'nt find the correct file, you could do:
`grep -r "to hide this message" /etc/`
This will show you the correct filename you'd like to change the content off.

**Comment by ghost** [comment]
> Try to edit `/etc/motd` or `/etc/update-motd.d/*` files.

I looked into the /etc/motd but its empty. And in update-motd.d, there are 3 Files called 10-uname, 85, 92-unattended-upgrades.

The 10-uname contains:

```
#!/bin/sh
uname -snrvm
```

And the 85-fwupd contains:

```
#!/bin/sh

if [ -f /run/motd.d/fwupd/85-fwupd ]; then
        cat /run/motd.d/fwupd/85-fwupd
fi
```

And the Folder "/run/motd.d" doesn't exists


And the 92-unattended-upgrades contains:
```
#!/bin/sh

if [ -x /usr/share/unattended-upgrades/update-motd-unattended-upgrades ]; then
    exec /usr/share/unattended-upgrades/update-motd-unattended-upgrades
fi


```

**Comment by s1ryx** [comment]
Found an entry in ``/etc/profile.d/kali.sh`` calling ``/usr/bin/kali-motd`` which is the script producing the output:
```bash
# Display message of the day information for cloud systems and
# minimum installs such as WSL or Docker
if [ -e /usr/bin/kali-motd ]; then
    kali-motd
fi
```
This also contains the message you want to edit:
```bash
┃ We have kept /usr/bin/python pointing to Python 2 for backwards
┃ compatibility. Learn how to change this and avoid this message:
┃ ⇒ https://www.kali.org/docs/general-use/python3-transition/
```
I removed the entry from profile.d alltogether cause I do not want to create a .hushlogin file for every user.

**Comment by TinnyTerr** [comment]
So I have found a roundabout way of doing it. Create a bash file of your motd (default for some dists is `/etc/update-motd.d/` but doesn't matter where if you aren't using it for anything else as long as you get the filepath correct.)

Open .bashrc and amend at the filepath to the end of the file with a optional comment. For example if I named one 01-custom (to use for ssh as well) in etc, I would add:
```bash
# MOTD for every login
/etc/update-motd.d/01-custom
```

This will appear at every login as .bashrc runs at every login. Comment is optional for future use and readability obviously. 
[This is the one I actively use](https://gist.github.com/TinnyTerr/13af5600219aab8678ef99406f385f22) as a start but it can be anything obviously
